pkg/kubectl/cmd/run.go

```diff
@@ -88,6 +88,14 @@ var (
 		kubectl run pi --schedule="0/5 * * * ?" --image=perl --restart=OnFailure -- perl -Mbignum=bpi -wle 'print bpi(2000)'`))
 )
 
+type RunObject struct {
+	Object  runtime.Object
+	Mapper  meta.RESTMapper
+	Mapping *meta.RESTMapping
+}
+
+var runObjectMap = map[string]*RunObject{}
```

Comment:
Thanks @soltysh ! I also didn't think global vars is a good solution. Will address.